standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/HiveAlterHandler.java

```diff
@@ -1031,61 +1029,28 @@ private Path constructRenamedPath(Path defaultNewPath, Path currentPath) {
-  }
-
   @VisibleForTesting
-  public static List<ColumnStatistics> deleteTableColumnStats(RawStore msdb, Table oldTable, Table newTable, List<ColumnStatistics> multiColStats)
+  public void deleteTableColumnStats(RawStore msdb, Table oldTable, Table newTable)
       throws InvalidObjectException, MetaException {
-    List<ColumnStatistics> newMultiColStats = new ArrayList<>();
     try {
       String catName = normalizeIdentifier(oldTable.isSetCatName()
           ? oldTable.getCatName()
           : getDefaultCatalog(msdb.getConf()));
       String dbName = oldTable.getDbName().toLowerCase();
       String tableName = normalizeIdentifier(oldTable.getTableName());
-      String newDbName = newTable.getDbName().toLowerCase();
-      String newTableName = normalizeIdentifier(newTable.getTableName());
       List<FieldSchema> oldTableCols = oldTable.getSd().getCols();
       List<FieldSchema> newTableCols = newTable.getSd().getCols();
-
-      boolean nameChanged = !newDbName.equals(dbName) || !newTableName.equals(tableName);
-
-      if ((nameChanged || !MetaStoreServerUtils.columnsIncludedByNameType(oldTableCols, newTableCols)) &&
+      if (!columnsIncludedByNameType(oldTableCols, newTableCols) ||
+          !MetaStoreServerUtils.arePrefixColumns(oldTableCols, newTableCols) &&
           // Don't bother in the case of ACID conversion.
           TxnUtils.isAcidTable(oldTable) == TxnUtils.isAcidTable(newTable)) {
-        for (ColumnStatistics colStats : multiColStats) {
-          List<ColumnStatisticsObj> statsObjs = colStats.getStatsObj();
-          List<ColumnStatisticsObj> newStatsObjs = new ArrayList<>();
-
-          if (statsObjs != null) {
-            for (ColumnStatisticsObj statsObj : statsObjs) {
-              boolean found = newTableCols.stream().anyMatch(c -> statsObj.getColName().equalsIgnoreCase(c.getName()) &&
-                  statsObj.getColType().equalsIgnoreCase(c.getType()));
-              if (nameChanged || !found) {
-                msdb.deleteTableColumnStatistics(catName, oldTable.getDbName().toLowerCase(),
-                    normalizeIdentifier(oldTable.getTableName()), statsObj.getColName(), colStats.getEngine());
-              }
-              if (found) {
-                newStatsObjs.add(statsObj);
-              }
-            }
-            StatsSetupConst.removeColumnStatsState(newTable.getParameters(),
-                statsObjs.stream().map(ColumnStatisticsObj::getColName).collect(Collectors.toList()));
-          }
-          ColumnStatisticsDesc statsDesc = colStats.getStatsDesc();
-          statsDesc.setDbName(newDbName);
-          statsDesc.setTableName(newTableName);
-          colStats.setStatsObj(newStatsObjs);
-          newMultiColStats.add(colStats);
+        List<String> deletedCols = oldTableCols.stream().filter(
+                c -> newTableCols.stream().noneMatch(n -> columnsIncludedByNameType(Arrays.asList(c), Arrays.asList(n))))
```

Comment:
Can we use a method to compare two columns excluding comment here instead of constructing arrays?